### PR TITLE
component reset/remove functionality

### PR DESCRIPTION
**Pull request by Tatev321**

### CODE REVIEW

File: editor/src/com/talosvfx/talos/editor/addons/scene/logic/components/SpineRendererComponent.java [hunk 1]
```diff
@@ -133,6 +134,11 @@ private void createSkeletonFromGameAsset () {
     @Override
     public void setGameAsset (GameAsset<SkeletonData> gameAsset) {
         this.gameAsset = gameAsset;
+
+        if(defaultGameAsset == null && !gameAsset.nameIdentifier.equals("broken")){
```
Comment (Tom-Ski) [inline editor/src/com/talosvfx/talos/editor/addons/scene/logic/components/SpineRendererComponent.java:138]: What is the purpose of this?

File: editor/src/com/talosvfx/talos/editor/addons/scene/logic/components/AComponent.java [hunk 2]
```diff
@@ -1,12 +1,29 @@
...
         return getClass();
     }
 
+    public void reset() {}
+
+    public void remove() {
+        if(gameObject!=null) {
+            gameObject.removeComponent(this);
+            SceneEditorWorkspace.getInstance().removeGizmos(gameObject);
+            SceneEditorWorkspace.getInstance().initGizmos(gameObject,  SceneEditorWorkspace.getInstance());
```
Comment (Tom-Ski) [inline editor/src/com/talosvfx/talos/editor/addons/scene/logic/components/AComponent.java:25]: Doesn't seem like this should be in in this component, rather where the remove is being called in the editor.

File: editor/src/com/talosvfx/talos/editor/addons/scene/logic/components/SpineRendererComponent.java (hunk 1 above)
Comment (Tatev321) [reply editor/src/com/talosvfx/talos/editor/addons/scene/logic/components/SpineRendererComponent.java:138]: when we import sprite at first we add broken then valid one

Comment (Tom-Ski) [reply editor/src/com/talosvfx/talos/editor/addons/scene/logic/components/SpineRendererComponent.java:138]: You are storing default for reset functionality?

**Review by Tom-Ski — APPROVED** [review]

**Pull request merged**